src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -179,6 +181,32 @@ def sort_code_mapping_configs(
                     sorted_configs.insert(0, config)
         return sorted_configs
 
+    def get_commit_sha(
+        self,
+        integration: Integration,
+        organization_id: str,
+        line_no: int,
+        filepath: str,
+        repository: Repository,
+        branch: str,
+    ) -> Optional[str]:
+        commitSha = None
+        integrationInstallation = integration.get_installation(organization_id=organization_id)
+        try:
+            git_blame_list = integrationInstallation.get_blame_for_file(
+                repository,
+                filepath,
+                branch,
+                line_no,
+            )
+            for blame in git_blame_list:
+                if blame["startingLine"] <= line_no <= blame["endingLine"]:
+                    commitSha = str(blame["commit"]["oid"])
+                    break
```

Comment:
Later in the code I call `capture_message`. Adding crumbs in here will be seen in there
```suggestion
                    break
            if not commitSha:
                logger.warning("Failed to get commit from git blame.", extra={"git_blame_response": git_blame_list, "filepath": filepath, "line_no": line_no, "branch": branch})
```